inlong-manager/manager-workflow/src/main/java/org/apache/inlong/manager/workflow/util/WorkflowUtils.java

```diff
@@ -0,0 +1,254 @@
+import org.slf4j.LoggerFactory;
+
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+
+/**
+ * Workflow utils
+ */
+public class WorkflowUtils {
+
+    public static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger LOGGER = LoggerFactory.getLogger(WorkflowUtils.class);
+
+    /**
+     * Build workflow context from WorkflowProcess and WorkflowProcessEntity
+     */
+    public static WorkflowContext buildContext(ObjectMapper objectMapper, WorkflowProcess process,
+            WorkflowProcessEntity processEntity) {
+        try {
+            ProcessForm form = WorkflowUtils.parseProcessForm(objectMapper, processEntity.getFormData(), process);
+            return new WorkflowContext().setProcess(process)
+                    .setOperator(processEntity.getApplicant())
+                    .setProcessForm(form)
+                    .setProcessEntity(processEntity);
+        } catch (Exception e) {
+            LOGGER.error("build context from process form failed with id=" + processEntity.getId(), e);
+            return null;
+        }
+    }
+
+    /**
+     * Get the workflow result from the given workflow context
+     */
+    public static WorkflowResult getResult(WorkflowContext context) {
+        if (context == null) {
+            return null;
+        }
+
+        WorkflowResult workflowResult = new WorkflowResult();
+        workflowResult.setProcessInfo(WorkflowUtils.getProcessResponse(context.getProcessEntity()));
+        if (context.getActionContext() != null) {
+            ActionContext newAction = context.getActionContext();
+            workflowResult.setNewTasks(Lists.newArrayList(WorkflowUtils.getTaskResponse(newAction.getTaskEntity())));
+        }
+        return workflowResult;
+    }
+
+    /**
+     * Get process response from process entity
+     */
+    public static ProcessResponse getProcessResponse(WorkflowProcessEntity entity) {
+        if (entity == null) {
+            return null;
+        }
+
+        ProcessResponse processResponse = ProcessResponse.builder()
+                .id(entity.getId())
+                .name(entity.getName())
+                .displayName(entity.getDisplayName())
+                .type(entity.getType())
+                .title(entity.getTitle())
+                .applicant(entity.getApplicant())
+                .status(ProcessStatus.valueOf(entity.getStatus()))
+                .startTime(entity.getStartTime())
+                .endTime(entity.getEndTime())
+                .build();
+        try {
+            if (StringUtils.isNotBlank(entity.getFormData())) {
+                processResponse.setFormData(OBJECT_MAPPER.readTree(entity.getFormData()));
+            }
+            if (StringUtils.isNotBlank(entity.getExtParams())) {
+                processResponse.setExtParams(OBJECT_MAPPER.readTree(entity.getExtParams()));
+            }
+        } catch (Exception e) {
+            LOGGER.error("parse process form error: ", e);
```

Comment:
It is strongly recommended to print entity when there is an error, which is convenient for troubleshooting

Reply:
The exception stack trace will print all info, such as:

<img width="1241" alt="image" src="https://user-images.githubusercontent.com/31994335/182747891-8010fd0d-71db-4360-a10d-b98f834c680a.png">